### DISCUSSION TITLE
Validity of comparing fp16 generated ESM2 embeddings with the ESM Atlas embeddings?

### DISCUSSION THREAD

**Discussion by allcatsaregrey** [facebookresearch/esm#684] [ESM Metagenomic Atlas]
If I am generating my own ESM2 embeddings locally, can I use fp16 precision during generation while still being able to make meaningful similarity comparisons to embeddings downloaded from the atlas? I see discussions suggesting people call .half() on the model to generate embeddings locally, but this does not exactly answer my question. Thank you for your time.

**Comment by ebetica** [comment]
Yes that's reasonable, the model was trained in fp16 precision. I think the results won't match exactly but will probably be good enough for most use cases (norm difference < 1e-3)
